### ISSUE TITLE
Expected a warning when no arguments are specified for a parameterized test method

### ISSUE THREAD

**Issue by jnm2** [nunit/nunit.analyzers#215]
There should be a warning that arguments are required.

The warning shouldn't be shown if an attribute is applied to the method that implements the same interface as `TestCaseAttribute` and `TestCaseSourceAttribute` or if an attribute is applied to each parameter that implements the same interface as `ValuesAttribute` and `ValueSourceAttribute`.

```cs
public class C
{
    [Test]
    public void M(int p)
    {
        Assert.That(p, Is.EqualTo(42));
    }
}
```

Do you think a fix could be nice? I had:

```cs
        [Test]
        public static void Examples_from_docs(string path, string encoded)
        {

        }
```

And maybe there could be a fix that does this:

```diff
-       [Test]
+       [TestCase("path", "encoded")]
        public static void Examples_from_docs(string path, string encoded)
        {

        }
```

**Comment by mikkelbu** [comment]
Notes:
* Iterate through the methods. If
  * the method has parameters and 
  * the method is marked with an attribute deriving from `ISimpleTestBuilder`
  * the method is not marked with an attribute deriving from `ITestBuilder` or at least one parameter is not marked by an attribute deriving from `IParameterDataSource`
* then report a problem

Somewhat related to #249
